cmd/manager/exec/manager.go

```diff
@@ -213,16 +211,42 @@ func RunManager() {
 			os.Exit(1)
 		}
 
+		hubKubeConfigCheckSum, err := utils.GetCheckSum(Options.HubConfigFilePathName)
+		if err != nil {
+			klog.Error("Failed to get the checksum of the hub kubeconfig file. ", err)
+			os.Exit(1)
+		}
+
 		leaseReconciler := leasectrl.LeaseReconciler{
-			HubKubeClient:        hubKubeClient,
-			KubeClient:           managedClusterKubeClient,
-			ClusterName:          Options.ClusterName,
-			LeaseName:            AddonName,
-			LeaseDurationSeconds: int32(Options.LeaseDurationSeconds),
+			HubKubeClient:         hubKubeClient,
+			HubConfigFilePathName: Options.HubConfigFilePathName,
+			HubConfigCheckSum:     hubKubeConfigCheckSum,
+			KubeClient:            managedClusterKubeClient,
+			ClusterName:           Options.ClusterName,
+			LeaseName:             AddonName,
+			LeaseDurationSeconds:  int32(Options.LeaseDurationSeconds),
 		}
 
-		go wait.JitterUntilWithContext(context.TODO(), leaseReconciler.Reconcile,
-			time.Duration(Options.LeaseDurationSeconds)*time.Second, leaseUpdateJitterFactor, true)
+		ctx, cancel := context.WithCancel(context.TODO())
+
+		go func() {
+			for {
+				err := leaseReconciler.CheckHubKubeConfig(ctx)
+				if err != nil {
+					cancel()
```

Comment:
What happens if this line does execute? Will it kill this goroutine and we lose this lease reconcile?

Reply:
Not only the goroutine, bu the whole controller will exit.  

the cancel() is called when there is kube kubeconfig change. Then it stops the context channel `ctx.Done()`.  As a result,  os.Exit is invoked to exit the whole controller for restart.